src/hil/spi_master.rs

```diff
@@ -1,73 +1,86 @@
-//! Traits and parameters for SPI master communication
+//! Traits and parameters for SPI master communication.
+//! Authors: Sam Crow <samcrow@uw.edu>
+//!          Philip Levis <pal@cs.stanfored.edu>
 
 use core::option::Option;
 
-#[derive(Copy, Clone)]
-pub enum Rate {
-    MSBFirst,
-    LSBFirst,
-}
 /// Values for the ordering of bits
 #[derive(Copy, Clone)]
-pub enum DataOrder {
-    MSBFirst,
-    LSBFirst,
-}
+pub enum DataOrder {MSBFirst, LSBFirst}
 
 /// Values for the clock polarity (idle state or CPOL)
 #[derive(Copy, Clone)]
-pub enum ClockPolarity {
-    IdleHigh,
-    IdleLow,
-}
-/// Values for the clock phase (CPHA), which defines when
-/// values are sampled
+pub enum ClockPolarity {IdleHigh, IdleLow}
+
+/// Which clock edge values are sampled on
 #[derive(Copy, Clone)]
-pub enum ClockPhase {
-    SampleLeading,
-    SampleTrailing,
-}
+pub enum ClockPhase {SampleLeading, SampleTrailing}
 
 pub trait SpiCallback {
     /// Called when a read/write operation finishes
-    fn read_write_done(&'static self, 
-                       read: Option<&'static mut[u8]>,
-                       writer: Option<&'static mut[u8]>);
```

Comment:
Hrm... don't we, in general, need these passed back in the callback? It works in your currect example code, because client's callback is wrapped in an unsafe block and just references the static variable directly every time. But that's not going to be possible in driver code.